polkadot/runtime/parachains/src/dmp.rs

```diff
@@ -226,6 +234,11 @@ impl<T: Config> Pallet<T> {
 			return Err(QueueDownwardMessageError::ExceedsMaxMessageSize)
 		}
 
+		// If the head exists, we assume the parachain is legit and exists.
+		if !paras::Heads::<T>::contains_key(para) {
+			return Err(QueueDownwardMessageError::Unroutable)
+		}
+
```

Comment:
Instead of duplicating these conditions from can_queue_downward_message, how about calling it directly here instead? e.g.:
```
let _ = Self::can_queue_downward_message(config, &para, &msg)?;
```

Reply:
Done